core/src/main/java/org/geysermc/geyser/network/UpstreamPacketHandler.java

```diff
@@ -68,10 +70,13 @@
 import java.util.ArrayDeque;
 import java.util.Deque;
 import java.util.OptionalInt;
+import java.util.HashMap;
 
 public class UpstreamPacketHandler extends LoggingPacketHandler {
 
-    private Deque<String> packsToSent = new ArrayDeque<>();
+    private final Deque<String> packsToSent = new ArrayDeque<>();
+
+    private PlayerResourcePackLoadEvent resourcePackLoadEvent;
```

Comment:
Any reason this is an instance variable?

Reply:
The packs of the event are also needed when the client requests a pack; therefore I'm storing them here for this instance.. unless there's some better way to get those packs from this fired event